### PR TITLE
Trace failed test cases on CI

### PR DESCRIPTION
**Pull request by overvenus**
It is sometimes difficult to debug a failed test without a detailed log. This PR brings log back to CI again.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -104,7 +104,7 @@ features = ["secure"]
 [profile.dev]
 opt-level = 0  # Controls the --opt-level the compiler builds with
 debug = true   # Controls whether the compiler passes `-g`
-codegen-units = 2
+codegen-units = 4
```
Comment (siddontang) [inline Cargo.toml]: why use 4 here?

File: ci-build/test.sh [hunk 2]
```diff
@@ -35,15 +35,10 @@ if [[ "$TRAVIS" = "true" ]]; then
 fi
 export RUSTFLAGS=-Dwarnings
 
-if [[ `uname` == "Linux" ]]; then
-    export EXTRA_CARGO_ARGS="-j 2"
```
Comment (siddontang) [inline ci-build/test.sh]: why remove it?

File: Cargo.toml (hunk 1 above)
Comment (overvenus) [reply Cargo.toml]: Both of them are trying to fix compilation failures in the Travis. See more https://github.com/pingcap/tikv/pull/1764

Comment (siddontang) [reply Cargo.toml]: oh, so you revert the change?

Comment (overvenus) [reply Cargo.toml]: Yes, tikv is no longer tested on the Travis anyway.

@BusyJay Do you have any suggestions?

**Comment by siddontang** [comment]
CI failed @overvenus

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by siddontang** [comment]
Ping @BusyJay

File: Cargo.toml (hunk 1 above)
Comment (BusyJay) [reply Cargo.toml]: Looks OK to me.

File: ci-build/test.sh [hunk 3]
```diff
@@ -35,15 +35,10 @@ if [[ "$TRAVIS" = "true" ]]; then
...
-fi
-
 if [[ "$SKIP_TESTS" != "true" ]]; then
     make test 2>&1 | tee tests.out
 else
-    export EXTRA_CARGO_ARGS="$EXTRA_CARGO_ARGS --no-run"
-    make test
+    EXTRA_CARGO_ARGS="--no-run" make test
```
Comment (BusyJay) [inline ci-build/test.sh]: Why remove the evaluation?

File: circle.yml [hunk 4]
```diff
@@ -58,11 +58,13 @@ dependencies:
   post:
     # check format first
     - make format && git diff-index --quiet HEAD -- || (git diff; echo please make format and run tests before creating a PR!; exit 1)
-    - cargo test --features "dev" --no-run:
+    - make check:
```
Comment (BusyJay) [inline circle.yml]: Isn't it making the test last longer than before?

**Comment by siddontang** [comment]
LGTM

**Pull request merged**